parlai/core/metrics.py

```diff
@@ -115,39 +127,157 @@ class AverageMetric(Metric):
     Class that keeps a running average of some metric.
     """
 
-    def __init__(self, numer: TScalar, denom: TScalar):
-        self._numer = self.as_float(numer)
-        self._denom = self.as_int(denom)
+    __slots__ = ('_numer', '_denom')
 
-    def __add__(self, other: 'AverageMetric') -> 'AverageMetric':
+    def __init__(self, numer: TScalar = 0, denom: TScalar = 1):
+        self._numer = self.as_number(numer)
+        self._denom = self.as_number(denom)
```

Comment:
Hmm - you want denom to be able to be a float now?

Reply:
No, I want it underspecified.